src/core/text/TextMetrics.js

```diff
@@ -0,0 +1,190 @@
+        this.lineWidths = lineWidths;
+        this.lineHeight = lineHeight;
+        this.maxLineWidth = maxLineWidth;
+        this.fontProperties = fontProperties;
+    }
+
+    /**
+     * Measures the supplied string of text and returns a Rectangle.
+     *
+     * @param {string} text - the text to measure.
+     * @param {PIXI.TextStyle} style - the text style to use for measuring
+     * @param {boolean} [wordWrap] - optional override for if word-wrap should be applied to the text.
+     * @param {HTMLCanvasElement} [canvas] - optional specification of the canvas to use for measuring.
+     * @return {PIXI.TextMetrics} measured width and height of the text.
+     */
+    static measure(text, style, wordWrap, canvas)
+    {
+        if (!canvas)
+        {
+            TextMetrics._canvas = TextMetrics._canvas || document.createElement('canvas');
+            canvas = TextMetrics._canvas;
+        }
+
+        wordWrap = wordWrap || style.wordWrap;
+        const font = Text.getFontStyle(style);
+        const fontProperties = Text.calculateFontProperties(font);
+        const context = canvas.getContext('2d');
+
+        context.font = font;
+
+        const outputText = wordWrap ? TextMetrics.wordWrap(text, style, canvas) : text;
+        const lines = outputText.split(/(?:\r\n|\r|\n)/);
+        const lineWidths = new Array(lines.length);
+        let maxLineWidth = 0;
+
+        for (let i = 0; i < lines.length; i++)
+        {
+            const lineWidth = context.measureText(lines[i]).width + ((lines[i].length - 1) * style.letterSpacing);
+
+            lineWidths[i] = lineWidth;
+            maxLineWidth = Math.max(maxLineWidth, lineWidth);
+        }
+        let width = maxLineWidth + style.strokeThickness;
+
+        if (style.dropShadow)
+        {
+            width += style.dropShadowDistance;
+        }
+
+        const lineHeight = style.lineHeight || fontProperties.fontSize + style.strokeThickness;
+        let height = Math.max(lineHeight, fontProperties.fontSize + style.strokeThickness)
+            + ((lines.length - 1) * lineHeight);
+
+        if (style.dropShadow)
+        {
+            height += style.dropShadowDistance;
+        }
+
+        return new TextMetrics(
+            text,
+            style,
+            width,
+            height,
+            lines,
+            lineWidths,
+            lineHeight,
+            maxLineWidth,
+            fontProperties
+        );
+    }
+
+    /**
+     * Applies newlines to a string to have it optimally fit into the horizontal
+     * bounds set by the Text object's wordWrapWidth property.
+     *
+     * @param {string} text - String to apply word wrapping to
+     * @param {PIXI.TextStyle} style - the style to use when wrapping
+     * @param {HTMLCanvasElement} [canvas] - optional specification of the canvas to use for measuring.
+     * @return {string} New string with new lines applied where required
+     */
+    static wordWrap(text, style, canvas)
```

Comment:
Why should `wordWrap` be private?  It's perfectly valid to use externally.  Currently, I don't have a use-case for that, but I could imagine that it might be useful.  No reason to hide something that *could* have potential benefits without an explicit reason to do so.